astropy/coordinates/baseframe.py

```diff
@@ -187,21 +187,30 @@ def _represent_as_dict(self):
         if coord.has_data:
             attrs.extend(coord.representation_component_names)
             # Don't output distance unless it's actually distance.
-            if isinstance(coord.data, r.UnitSphericalRepresentation):
+            if issubclass(
+                coord.get_representation_cls(), r.SphericalRepresentation
+            ) and isinstance(coord.data, r.UnitSphericalRepresentation):
                 attrs = attrs[:-1]
 
             diff = coord.data.differentials.get("s")
             if diff is not None:
                 diff_attrs = list(coord.get_representation_component_names("s"))
-                # Don't output proper motions if they haven't been specified.
-                if isinstance(diff, r.RadialDifferential):
-                    diff_attrs = diff_attrs[2:]
-                    # Don't output radial velocity unless it's actually velocity.
-                elif isinstance(
-                    diff,
-                    (r.UnitSphericalDifferential, r.UnitSphericalCosLatDifferential),
+                if issubclass(
+                    coord.get_representation_cls("s"),
+                    (r.SphericalDifferential, r.SphericalCosLatDifferential),
                 ):
-                    diff_attrs = diff_attrs[:-1]
+                    # Don't output proper motions or radial velocities unless
+                    # they've been specifically passed in.
+                    if isinstance(diff, r.RadialDifferential):
+                        diff_attrs = diff_attrs[2:]
+                    elif isinstance(
+                        diff,
+                        (
+                            r.UnitSphericalDifferential,
+                            r.UnitSphericalCosLatDifferential,
+                        ),
+                    ):
+                        diff_attrs = diff_attrs[:-1]
```

Comment:
Slicing a `list` forces making a copy:
```suggestion
                        diff_attrs.pop()
```

Reply:
I can do that, but prefer not to, as I think the code is clearer if it uses the slices consistently (and this is not performance critical).